apps/framework-cli/src/cli/display/infrastructure.rs

```diff
@@ -15,6 +16,113 @@ use crate::framework::core::{
 use crossterm::{execute, style::Print};
 use log::info;
 
+/// Create the detail indentation string at compile time
+/// Computed from ACTION_WIDTH (15) + 3 spaces:
+/// - ACTION_WIDTH spaces for the action column
+/// - 1 space after the action symbol (e.g., "+", "-", "~")  
+/// - 2 spaces for additional indentation of detail lines
+///   Total: 18 spaces for proper alignment
+const DETAIL_INDENT: &str = {
+    const INDENT_BYTES: [u8; ACTION_WIDTH + 3] = [b' '; ACTION_WIDTH + 3];
+    // This is safe because we know all bytes are spaces (0x20), which are valid UTF-8
+    match std::str::from_utf8(&INDENT_BYTES) {
+        Ok(s) => s,
+        Err(_) => panic!("Invalid UTF-8 in DETAIL_INDENT"),
+    }
+};
+
+/// Helper function to write detail lines with proper indentation
+fn write_detail_lines(details: &[String]) {
+    let mut stdout = std::io::stdout();
+    for detail in details {
+        execute!(stdout, Print(DETAIL_INDENT), Print(detail), Print("\n"))
+            .expect("failed to write detail to terminal");
+    }
+}
+
+/// Formats a column type for display, showing enum values if it's an enum
+fn format_column_type(col_type: &ColumnType) -> String {
+    match col_type {
+        ColumnType::Enum(data_enum) => {
+            let values = data_enum
+                .values
+                .iter()
+                .map(|member| match &member.value {
+                    EnumValue::Int(i) => format!("{}={}", member.name, i),
+                    EnumValue::String(s) => format!("{}='{}'", member.name, s),
+                })
+                .collect::<Vec<_>>()
+                .join(",");
+            format!("Enum<{}>[{}]", data_enum.name, values)
+        }
+        _ => col_type.to_string(),
+    }
+}
+
+/// Formats a table for display with columns shown vertically
+/// Returns a tuple of (title, details) for use with detailed display functions
+fn format_table_display(
+    table: &crate::framework::core::infrastructure::table::Table,
+) -> (String, Vec<String>) {
+    let mut details = Vec::new();
+
+    // Table header
+    let title = if let Some(ref version) = table.version {
+        format!("Table: {} (Version: {})", table.name, version)
+    } else {
+        format!("Table: {}", table.name)
+    };
+
+    // Columns section
+    details.push("Columns:".to_string());
+    for column in &table.columns {
+        let mut col_str = format!(
+            "  {}: {}",
+            column.name,
+            format_column_type(&column.data_type)
+        );
+
+        // Add nullability
+        if column.required {
+            col_str.push_str(" NOT NULL");
+        } else {
+            col_str.push_str(" NULLABLE");
+        }
```

Comment:
i don't like that this mixes SQL syntax with the implementation `Display for ColumnType` strings